src/front/wgsl/mod.rs

```diff
@@ -452,24 +454,35 @@ impl BindingParser {
                 lexer.expect(Token::Paren('('))?;
                 let raw = lexer.next_ident()?;
                 self.interpolation = Some(conv::map_interpolation(raw)?);
-                lexer.expect(Token::Paren(')'))?;
+                match lexer.next() {
+                    (Token::Separator(','), _) => {
+                        let raw = lexer.next_ident()?;
+                        self.sampling = Some(conv::map_sampling(raw)?);
+                        lexer.expect(Token::Paren(')'))?;
+                    }
+                    (Token::Paren(')'), _) => (),
+                    other => {
+                        return Err(Error::Unexpected(other, "'interpolate' attribute value"));
+                    }
+                }
             }
             _ => return Err(Error::UnknownAttribute(name)),
         }
         Ok(())
     }
 
     fn finish<'a>(self) -> Result<Option<crate::Binding>, Error<'a>> {
-        match (self.location, self.built_in, self.interpolation) {
-            (None, None, None) => Ok(None),
-            (Some(loc), None, interpolation) => {
-                Ok(Some(crate::Binding::Location(loc, interpolation)))
+        match (self.location, self.built_in, self.interpolation, self.sampling) {
```

Comment:
There is a discussion to be had on whether the IR can accept `None` for interpolation where it's expected to be, i.e. whether IR should have a notion of the default interpolation. I think the IR is stronger if it doesn't have defaults. 

That is, whenever IR sees that an interpolation would be needed (i.e. vertex outputs and fragment inputs), it should expect it to be `Some` and error otherwise. That means, in turn that all the frontends would need to resolve their "default" semantics, and it makes sense to me given that it's *their* default to resolve. I.e. the defaults can be different in frontends. For example SPIR-V probably always defaults to "Perspective", while WGSL defaults to "Flat" for integers:
> User-defined IO of scalar or vector integer type is always [[interpolate(flat)]] and, therefore, must not be specified in a WGSL program.

Resolving the default interpolation may be a little bit annoying, since it's probably the code that all frontends would need to use, similarly: scan all entry points, scan all the IO structures they use, check all the locations on them. So I think we'd need a small processor, e.g. `src/proc/interpolator.rs`, that frontends would run before producing a module.